k8s/controller.yaml

```diff
@@ -48,15 +48,14 @@ spec:
 
 ---
 
-kind: StatefulSet
+kind: Deployment
```

Comment:
To provide some more context, the original discussion regarding StatefulSet vs. Deployment is here: https://github.com/kubernetes-sigs/agent-sandbox/pull/13#discussion_r2356834262

Note that if we decide to accept this change, this requires an additional migration step when upgrading Agent Sandbox to this version.